nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ProcessGroupStatusSnapshotDTO.java

```diff
@@ -563,28 +563,66 @@ public ProcessGroupStatusSnapshotDTO clone() {
 
         other.setProcessingNanos(getProcessingNanos());
 
-        other.setConnectionStatusSnapshots(copy(getConnectionStatusSnapshots()));
-        other.setProcessorStatusSnapshots(copy(getProcessorStatusSnapshots()));
-        other.setRemoteProcessGroupStatusSnapshots(copy(getRemoteProcessGroupStatusSnapshots()));
-        other.setInputPortStatusSnapshots(copy(getInputPortStatusSnapshots()));
-        other.setOutputPortStatusSnapshots(copy(getOutputPortStatusSnapshots()));
+        if (getConnectionStatusSnapshots() != null) {
```

Comment:
Using the `get` method here and in the for loop seems unnecessary when the member variable is directly accessible in this method.